source/ranges.tex

```diff
@@ -1436,6 +1439,48 @@
 \end{codeblock}
 \end{itemdescr}
 
+\rSec2[range.dangling]{Dangling iterator handling}
+
+\pnum
+The tag type \tcode{dangling} is used together
+with the template aliases \tcode{safe_iterator_t} and \tcode{safe_subrange_t}
+to indicate that an algorithm
+that typically returns an iterator into or subrange of a \tcode{Range} argument
+does not return an iterator or subrange
+which could potentially reference a range
+whose lifetime has ended for a particular rvalue \tcode{Range} argument
+which does not model \tcode{\placeholder{forwarding-range}}\iref{range.range}.
+\begin{codeblock}
+namespace std {
+  struct dangling {
+    constexpr dangling() noexcept = default;
+  template<class... Args>
+    constexpr dangling(Args&&...) noexcept { }
```

Comment:
The above two lines need an additional indent.

Reply:
Fixed.